File: src/atcoder/abc314/swift/abc314_b.swift
let n = Int(readLine()!)!

struct Player {
  let id:Int
  let num:Int
  let bet:[Int]
}

var players = [Player]()
for i in 0..<n {
  let c = Int(readLine()!)!
  let a = readLine()!.split(separator: " ").map { Int($0)! }
  players.append(
    Player(id: i, num: c, bet: a)
  )
}

let x = Int(readLine()!)!

let winner = players.filter {
  $0.bet.contains(x)
}

if winner.isEmpty {
  print(0)
} else {
  let mini = winner.map{$0.num}.min()
  let ans = winner.filter {
    $0.num == mini
  }.map {
    String($0.id + 1)
  } 
  print(ans.count)
  print(ans.joined(separator: " "))
}

File: src/atcoder/abc314/scala/abc314_b.scala
import io.StdIn.readLine
import scala.collection.mutable.ArrayBuffer

case class Player(id: Int, num: Int, bet: Array[Int])

object Main extends App:
  val n = readLine.toInt
  var players = ArrayBuffer[Player]()

  for i <- 0 until n do
    val c = readLine.toInt
    val a = readLine.split(" ").map(_.toInt)
    players += Player(i + 1, c, a)

  val x = readLine.toInt

  val winner = players.filter(_.bet.contains(x))

  if winner.isEmpty then
    println(0)
  else
    val mini = winner.map(_.num).min
    val ans = winner.filter(_.num == mini)
    println(ans.size)
    println(ans.map(_.id).mkString(" "))
